### DISCUSSION TITLE
Confusion about layouts and routes

### DISCUSSION THREAD

**Discussion by djscruggs** [remix-run/remix#8584] [Q&A] [answered]
I have an app that allows users to create challenges, like meditating every day for 30 days. My url structure looks like this

```
/challenges - index
/challenges/new - form for new challenge
/challenges/$challengeId - see a specific challenge
/challenges/$challengeId/checkin - check in to show you did the activity
```

After watching lots of videos and reading, my understanding is this folder structure should work

```
/challenges.tsx -- layout for challenges
/challenges/index.tsx -- index page
/challenges/new.tsx -- new challenge form
/challenges/$challengeId.tsx -- see a specific challenge
/challenges/$challengeId/checkin -- checkin page

```
However, none of this is working for me. `challenges/index.tsx` loads the home page, but it doesn't have the layout. `challenges/new` doesn't load at all

**Comment by sergiodxa** [comment, marked as answer]
This files:
```
/challenges.tsx -- layout for challenges
/challenges/index.tsx -- index page
/challenges/new.tsx -- new challenge form
/challenges/$challengeId.tsx -- see a specific challenge
/challenges/$challengeId/checkin -- checkin page
```
Worked on the v1 of Remix, but since v2 the file convention for routes changed to be flat, you can achieve the same with:
```
routes/challenges.tsx -- layout for challenges
routes/challenges._index.tsx -- index page
routes/challenges.new.tsx -- new challenge form
routes/challenges.$challengeId.tsx -- see a specific challenge
routes/challenges.$challengeId.checkin.tsx -- checkin page
```
Noe that this is supposing everything is nested, since mention the first challenges.tsx as the layout I imagine everything is instead inside that one, but if let's say `routes/challenges.$challengeId.checkin.tsx` is not nested in the UI of `routes/challenges.$challengeId.tsx`, then the file should be `routes/challenges.$challengeId_.checkin.tsx`.

You can read everything about the convention here https://remix.run/docs/en/main/file-conventions/routes.

**Reply by djscruggs** [reply]
So everything has to be in one directory going forward? Doesn't seem optimal

Also, maybe [this video](https://remix.guide/resources/Ws68TSouit76) should be taken down. Not a lot comes up when I search for routes. Just the official guide (which doesn't include standard CRUD routes), a couple github issues and that video.

**Reply by sergiodxa** [reply]
> So everything has to be in one directory going forward? Doesn't seem optimal

Yes, there are many reasons for this, you can check https://github.com/remix-run/remix/discussions/4482 to see the reason behind this change.

> Also, maybe [this video](https://remix.guide/resources/Ws68TSouit76) should be taken down

That video is not from the official Remix team, you would have to ask the author, maybe just adding a comment saying it's for Remix v1 is enough.

> Not a lot comes up when I search for routes. Just the official guide (which doesn't include standard CRUD routes), a couple github issues and that video

A lot of discussions about this happened on the [Discord server](https://rmx.as/discord), there you will find way more content.
